# Disallow when a DOM element is using both children and dangerouslySetInnerHTML' (`react-ts/no-dangerously-set-innerhtml`)

ğŸ’¼ This rule is enabled in the following configs: âœ… `recommended`, `recommended-type-checked`.

<!-- end auto-generated rule header -->

Disallow when a DOM element is using both children and `dangerouslySetInnerHTML`

## Rule Details

### âŒ Incorrect

```tsx
function Component() {
    return <div dangerouslySetInnerHTML={{ __html: "Hello World" }}>Hello World</div>;
}
```
