# Disallow when a DOM element is using both children and dangerouslySetInnerHTML' (`react-ts/no-dangerously-set-innerhtml-with-children`)

ğŸ’¼ This rule is enabled in the following configs: â˜‘ï¸ `recommended`, `recommended-type-checked`.

<!-- end auto-generated rule header -->

Disallow when a DOM element is using both children and `dangerouslySetInnerHTML`

## Rule Details

### âŒ Incorrect

```tsx
<div dangerouslySetInnerHTML={{ __html: "Hello World" }}>Hello World</div>;
```

### âœ… Correct

```tsx
<div dangerouslySetInnerHTML={{ __html: "Hello World" }} />;
```
