# no-dangerously-set-innerhtml-with-children

## Rule category

Correctness.

## What it does

Disallows `DOM element` using `children` and `dangerouslySetInnerHTML` at the same time.

## Why is this bad?

When using `dangerouslySetInnerHTML`, the content of the DOM element is set from the `__html` property. The content of the DOM element is completely replaced, so the children will not be rendered as expected.

## Examples

### Failing

```tsx twoslash
import React from "react";

function Example() {
  return (
    <div dangerouslySetInnerHTML={{ __html: "Hello World" }}>
// @error:     ðŸ‘† Do not use `dangerouslySetInnerHTML` and `children` at the same time
      <p>Goodbye World</p>
    </div>
  );
}
```

### Passing

```tsx twoslash
import React from "react";

function Example() {
  return <div dangerouslySetInnerHTML={{ __html: "Hello World" }} />;
}
```
