# no-clone-element

## Rule category

Restriction.

## What it does

Prevents the use of `React.cloneElement`.

## Why is this bad?

Using cloneElement is uncommon and can lead to fragile code. This also makes it harder to trace the data flow. Try the [alternatives](https://react.dev/reference/react/cloneElement#alternatives) instead.

## Examples

### Failing

```tsx
import { cloneElement } from "react";

// ...
const clonedElement = cloneElement(
  // Using 'cloneElement' is uncommon and can lead to fragile code. Use alternatives instead.
  <Row title="Cabbage">Hello</Row>,
  { isHighlighted: true },
  "Goodbye"
);
// ...

console.log(clonedElement); // <Row title="Cabbage" isHighlighted={true}>Goodbye</Row>
```

## Further Reading

- [react.dev: Legacy React APIs cloneElement](https://react.dev/reference/react/createRef)
