# no-deprecated-string-refs

ğŸ’¼ This rule is enabled in the ğŸ‘ `recommended` config.

<!-- end auto-generated rule header -->

Disallow using deprecated string refs

## Rule Details

This rule disallows using deprecated string refs.

### âŒ Incorrect

```tsx
function Component() {
    return <div ref="example" />;
}
```

### âœ… Correct

```tsx
function Component() {
    const ref = useRef<HTMLDivElement>(null);

    return <div ref={ref} />;
}
```
