# jsx/no-script-url

ğŸ’¼ This rule is enabled in the ğŸ‘ `recommended` config.

<!-- end auto-generated rule header -->

Prevents usage of `javascript:` URLs as the value of the `href` prop in JSX.

## Rule Details

### âŒ Incorrect

```tsx
<a href="javascript:"></a>
<a href="javascript:void(0)"></a>
<a href="j\n\n\na\rv\tascript:"></a>
```

### âœ… Correct

```tsx
<Foo href="javascript:"></Foo>
<a href={"javascript:"}></a>
<a href="https://example.com"></a>
```
