# jsx/prefer-shorthand-boolean

âš ï¸ This rule _warns_ in the following configs: ğŸ¨ `jsx`, ğŸ‘ `recommended`.

<!-- end auto-generated rule header -->

## Rule Details

This rule enforces the use of shorthand syntax for boolean attributes.

### âŒ Incorrect

```tsx
<Component hidden={true} disabled={false} />;
```

### âœ… Correct

```tsx
<Component hidden disabled />;
```
