---
title: no-useless-custom-hooks
---

**Full Name in `eslint-plugin-react-hooks-extra`**

```plain copy
react-hooks-extra/no-useless-custom-hooks
```

**Full Name in `@eslint-react/eslint-plugin`**

```plain copy
@eslint-react/hooks-extra/no-useless-custom-hooks
```

**Features**

`üîç`

**Presets**

- `recommended`
- `recommended-typescript`
- `recommended-type-checked`

## What it does

Enforces custom Hooks to use at least one other Hook inside.

If your function doesn‚Äôt call any Hooks, avoid the `use` prefix. Instead, write it as a regular function without the `use` prefix. For example, `useSorted` below doesn‚Äôt call Hooks, so call it `getSorted` instead:

```tsx
// üî¥ Avoid: A Hook that doesn't use Hooks
function useSorted(items) {
  return items.slice().sort();
}
```

```tsx
// ‚úÖ Good: A regular function that doesn't use Hooks
function getSorted(items) {
  return items.slice().sort();
}
```

This ensures that your code can call this regular function anywhere, including conditions:

```tsx
function List({ items, shouldSort }) {
  let displayedItems = items;
  if (shouldSort) {
    // ‚úÖ It's ok to call getSorted() conditionally because it's not a Hook
    displayedItems = getSorted(items);
  }
  // ...
}
```

You should give `use` prefix to a function (and thus make it a Hook) if it uses at least one Hook inside of it:

```tsx
// ‚úÖ Good: A Hook that uses other Hooks
function useAuth() {
  return useContext(Auth);
}
```

Technically, this isn‚Äôt enforced by React. In principle, you could make a Hook that doesn‚Äôt call other Hooks. This is often confusing and limiting so it‚Äôs best to avoid that pattern. However, there may be rare cases where it is helpful. For example, maybe your function doesn‚Äôt use any Hooks right now, but you plan to add some Hook calls to it in the future. Then it makes sense to name it with the `use` prefix:

```tsx
// ‚úÖ Good: A Hook that will likely use some other Hooks later
function useAuth() {
  // TODO: Replace with this line when authentication is implemented:
  // return useContext(Auth);
  return TEST_USER;
}
```

Then components won‚Äôt be able to call it conditionally. This will become important when you actually add Hook calls inside. If you don‚Äôt plan to use Hooks inside it (now or later), don‚Äôt make it a Hook.

## Examples

### Failing

```tsx
function useSorted(items) {
  return items.slice().sort();
}
```

```tsx
// No 'TODO' and 'useContext()' comments inside function body
function useAuth() {
  return TEST_USER;
}
```

### Passing

```tsx
function getSorted(items) {
  return items.slice().sort();
}
```

```tsx
function useAuth() {
  return useContext(Auth);
}
```

```tsx
function useAuth() {
  // TODO: Replace with this line when authentication is implemented:
  // return useContext(Auth);
  return TEST_USER;
}
```

## Implementation

- [Rule source](https://github.com/Rel1cx/eslint-react/tree/main/packages/plugins/eslint-plugin-react-hooks-extra/src/rules/no-useless-custom-hooks.ts)
- [Test source](https://github.com/Rel1cx/eslint-react/tree/main/packages/plugins/eslint-plugin-react-hooks-extra/src/rules/no-useless-custom-hooks.spec.ts)

## Further Reading

- [React: invalid-hook-call-warning (the first note)](https://react.dev/warnings/invalid-hook-call-warning)
- [React: should-all-functions-called-during-rendering-start-with-the-use-prefix (the deep dive)](https://react.dev/learn/reusing-logic-with-custom-hooks#should-all-functions-called-during-rendering-start-with-the-use-prefix)
