# no-missing-iframe-sandbox

## What it does

Enforces explicit `sandbox` attribute for `iframe` elements.

## Why is this bad?

The sandbox attribute enables an extra set of restrictions for the content in the iframe. Using sandbox attribute is considered a good security practice.

## Examples

This rule checks all React iframe elements and verifies that there is sandbox attribute and that it's value is valid.

### Failing

```tsx
import React from "react";

function Example() {
  return <iframe src="https://example.com" />;
  //     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  //     - Missing 'sandbox' attribute on iframe component.
}
```

### Passing

```tsx
import React from "react";

function Example() {
  return <iframe src="https://example.com" sandbox="allow-popups" />;
}
```

## Further reading

- [MDN: iframe - sandbox](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#attributes)
